### ISSUE TITLE
Make use of JDK9+ features

### ISSUE THREAD

**Issue by acoburn** [trellis-ldp/trellis#106]
There are a number of `TODO` comments in the code, indicating places where newer, JDK9+ features can be used. This is a placeholder ticket for making those changes.

Depends on #66 and on changing the minimum Java requirement from 8 to 11.

These are some of the new features (in addition to JPMS) that will make the code simpler:

  * factory methods for `Collection` types (e.g. `List::of`)
  * untyped local variables (`var` syntax, as of JDK 10)
  * expanded `Optional` interface (particularly `::stream` and `::or`)
  * expanded `InputStream` interface (particularly `::transferTo`)

**Comment by acoburn** [comment]
Closing this, as it is, basically, a duplicate of #66
